lib/Generate.js

```diff
@@ -59,110 +61,51 @@ function addKernelTestChecking(content) {
   );
 }
 
-function generateElmJson(
-  projectRootDir /*: string */,
-  generatedCodeDir /*: string */,
-  hasBeenGivenCustomGlobs /*: boolean */,
-  elmJsonPath /*: string */,
-  projectElmJson /*: any */
-) /*: [string, Array<string>] */ {
-  const testRootDir = Compile.getTestRootDir(projectRootDir);
-  const generatedSrc = path.join(generatedCodeDir, 'src');
-
-  var isPackageProject = projectElmJson.type === 'package';
-
-  // if we were given file globs, we don't need to check the tests/ directory exists
-  // this is only for elm applications, which people may need to introduce slowly into their apps
-  // for packages, we stick with the existing behaviour and assume tests are in tests/ so do the check always
-  const needToCareAboutTestsDir =
-    hasBeenGivenCustomGlobs === false || isPackageProject === true;
-
-  // we add the tests dir as a source if:
-  // - we decided above we need to care about it
-  // - if it exists on disk: this supports the case where we have all our tests in tests/ but
-  //   want to pass a glob to only run one of the test files
-  const shouldAddTestsDirAsSource =
-    needToCareAboutTestsDir ||
-    fs.existsSync(path.join(projectRootDir, 'tests'));
```

Comment:
This has been moved to Project.js, and has been greatly simplified. All we need to do is – if tests/ exists, then add it as a source directory, otherwise don’t, regardless of whether testing a package or an application.

Reply:
sounds good!